_source/_posts/2022-09-08-react-fastify-postgres.md

```diff
@@ -0,0 +1,917 @@
+  Visited: boolean;
+  id: bigint;
+}
+
+async function facilitiesRoutes(fastify: FastifyInstance) {
+  const client = await fastify.pg.connect();
+
+  fastify.get(
+    "/facilities",
+    async (request: FastifyRequest, reply: FastifyReply) => {
+      let facilities: IFacility[] = [];
+
+      try {
+        const { rows } = await client.query("SELECT * FROM facilities");
+        if (rows.length == 0) throw new Error("No facilities found");
+
+        facilities = rows;
+        reply
+          .code(200)
+          .header("Content-Type", "application/json; charset=utf-8")
+          .send([...facilities]);
+      } catch (error) {
+        const errorMessage = (error as Error).message;
+        throw new Error(errorMessage);
+      }
+    }
+  );
+
+  fastify.patch(
+    "/facilities/:id",
+    async (request: FastifyRequest, reply: FastifyReply) => {
+      const { id } = request.params as { id: bigint };
+      const { visited } = request.body as { visited: boolean };
+
+      const query = {
+        text: `UPDATE public.facilities SET 
+                "Visited" = COALESCE($1, "Visited")
+                WHERE id = $2`,
+        values: [visited, id],
+      };
+
+      try {
+        await client.query(query);
+        reply.code(204);
+      } catch (error) {
+        const errorMessage = (error as Error).message;
+        throw new Error(errorMessage);
+      }
+    }
+  );
+
+  fastify.delete(
+    "/facilities/:id",
+    async (request: FastifyRequest, reply: FastifyReply) => {
+      const { id } = request.params as { id: bigint };
+
+      const query = {
+        text: `DELETE FROM public.facilities
+                  WHERE id = $1 RETURNING *`,
+        values: [id],
+      };
+
+      try {
+        await client.query(query);
+        reply.code(204);
+      } catch (error) {
+        const errorMessage = (error as Error).message;
+        throw new Error(errorMessage);
+      }
+    }
+  );
+}
+
+export default facilitiesRoutes;
+
+```
+
+{:start="7"}
+7. To register these routes with our Fastify instance, add the following to `index.ts`:
+```ts
+fastify.register(facilitiesRoutes);
+```
+
+**NOTE:** Don't forget to also import the `facilitiesRoutes` after adding the above line.
+
+{:start="8"}
+8. Then we'll create a `utils/jwt-verifier.ts` file that will include logic to verify the Okta ID tokens included in API calls from the frontend:
+
+```ts
+import OktaJwtVerifier from "@okta/jwt-verifier";
+import dotenv from "dotenv";
+import { FastifyReply, FastifyRequest } from "fastify";
+
+dotenv.config();
+
+const oktaJwtVerifier = new OktaJwtVerifier({
+  issuer: process.env.OKTA_ISSUER || "",
+  clientId: process.env.OKTA_CLIENT_ID,
+});
+
+const audience = process.env.OKTA_AUDIENCE;
+
+export const jwtVerifier = async (
+  request: FastifyRequest,
+  reply: FastifyReply
+) => {
+  const { authorization } = request.headers;
+
+  request.log.info({ authorization });
+
+  const [token] = authorization?.trim().split(" ") || "";
```

Comment:
oh! This should have `Bearer` prepending the token. Would it be possible to change this (and update the code in tutorial/sample)? At least it looks like it's missing looking at this. Feel free to correct me.

Reply:
@alisaduncan I had it like this initially, but the `jwtVerifier` on the backend looks like it's expecting only the token without `Bearer `. Will I just need to remove the `Bearer ` when doing the JWT verify on the backend?